presto-native-execution/presto_cpp/main/PrestoServer.cpp

```diff
@@ -945,6 +953,49 @@ void PrestoServer::addServerPeriodicTasks() {
         ttlCheckInterval,
         "cache_ttl");
   }
+
+  if (isCachePeriodicFullPersistenceEnabled()) {
+    PRESTO_STARTUP_LOG(INFO)
+        << "Initializing cache periodic full persistence task...";
+    const int64_t cacheFullPersistenceInterval =
+        std::chrono::duration_cast<std::chrono::seconds>(
+            SystemConfig::instance()->asyncCacheFullPersistenceInterval())
+            .count();
+    const auto asyncCacheSsdCheckpointGb =
+        SystemConfig::instance()->asyncCacheSsdCheckpointGb();
+    periodicTaskManager_->addTask(
+        [asyncCacheSsdCheckpointGb]() {
+          auto* cache = velox::cache::AsyncDataCache::getInstance();
+          if (cache == nullptr) {
+            return;
+          }
+
+          auto* ssdCache = cache->ssdCache();
+          if (ssdCache == nullptr) {
+            return;
+          }
```

Comment:
If cache and ssdcache both are non-null, shall we just ignore it instead of failing the server hard? Similarly, we still allow the server operation api to be called even when there is no cache.

Reply:
What I meant is something like this
```
if (isCachePeriodicFullPersistenceEnabled()) {
  VELOX_CHECK_NOT_NULL(cache);
  VELOX_CHECK_NOT_NULL(ssdCache);
}
```
Or are there any scenarios that they can actually be NULL when the if condition is true?